code.py

```diff
@@ -53,12 +56,33 @@ def nextLine(self, pose):
 
 
 	# Function D in paper: returns dist b/w oldpose and next line
-	def nextDist(self, pose, line):
+	def nextDist(self, pose, line, index_bot):
+		x0 = pose[0]
+		x1 = pose[1]
+		x1 = line[0][0]
+		y1 = line[0][1]
+		x2 = line[1][0]
+		y2 = line[1][1]
+		
+		d1 = math.sqrt((x1 - x0)**2 + (y1 - y0)**2)
+		d2 = math.sqrt((x2 - x0)**2 + (y2 - y0)**2)
+		
+		if d1 < d2:
+			self.pose[index_bot] = [x2, y2]
```

Comment:
You can return d1 here rather than finding min() again later.

Reply:
Yeah,... will do